File: CppPractice/Project1/Project1/Leetcode121.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;


class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size()<=1)
        {
            return 0;

        }
        int ans = 0;
        int temp_min = prices[i]];
        for (int i = 1; i < prices.size(); i++)
        {
            if (prices[i] <temp_min)
            {
                temp_min = prices[i];

            }
            ans = max(ans, prices[i] - temp_min);

        }

            
        return ans;
    }
};

File: Pythonleetcode/Leetcode121.py
class Solution(object):
    def maxProfit(self, prices):

        if len(prices) == 0:
            return 0

        maxProfit = 0
        minPrice = prices[0]

        for i in range(len(prices)):
            if prices[i] < minPrice:
                minPrice = prices[i]
            if prices[i] - minPrice > maxProfit:
                maxProfit = prices[i] - minPrice

        return maxProfit







print(Solution().maxProfit([1,2,11,0,4,7]))